bicep/main.bicep

```diff
@@ -0,0 +1,329 @@
+param SAFARICOM_API_URL string
+param SAFARICOM_B2C_PAYMENTREQUEST_ENDPOINT string
+@secure()
+param SAFARICOM_CONSUMER_KEY string
+@secure()
+param SAFARICOM_CONSUMER_SECRET string
+param SAFARICOM_IDTYPE string
+param SAFARICOM_INITIATORNAME string
+param SAFARICOM_PARTY_A string
+@secure()
+param SAFARICOM_SECURITY_CREDENTIAL string
+@secure()
+param TWILIO_AUTHTOKEN string
+param TWILIO_MESSAGING_SID string
+param TWILIO_SID string
+param TWILIO_WHATSAPP_NUMBER string
+param USERCONFIG_121_SERVICE_EMAIL_ADMIN string
+param WEBSITE_HTTPLOGGING_RETENTION_DAYS string
+param WEBSITE_RUN_FROM_PACKAGE string
+param WEBSITES_CONTAINER_START_TIME_LIMIT string
+param XDT_MicrosoftApplicationInsights_Mode string
+
+
+// Find existing logAnalyticsWorkspace
+resource logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@2023-09-01' existing = {
+  name: logAnalyticsWorkspaceName
+}
+
+// Create Application Insights resource
+resource appInsights 'Microsoft.Insights/components@2020-02-02' = {
+  name: '121-${clientName}'
+  location: location
+  kind: 'web'
+  properties: {
+    Application_Type: 'web'
+    WorkspaceResourceId: logAnalyticsWorkspace.id
+  }
+  tags: tags
+}
+
+//Find existing App Service Plan
+resource appServicePlan 'Microsoft.Web/serverfarms@2022-03-01' existing = {
+  name: appServicePlanName
+}
+
+//Create new App Service
+resource newAppService 'Microsoft.Web/sites@2022-03-01' = {
+  name: appServiceName
+  location: location
+  kind: 'app,linux'
+  properties: {
+    serverFarmId: appServicePlan.id
+    siteConfig: {
+      appSettings: [
+        {
+          name: 'POSTGRES_PASSWORD'
+          value: POSTGRES_PASSWORD
+        },{
+          name: 'POSTGRES_USER'
+          value: POSTGRES_USER
+        },{
+          name: 'RESET_SECRET'
+          value: RESET_SECRET
+        },{
+          name: 'SECRETS_121_SERVICE_SECRET'
+          value: SECRETS_121_SERVICE_SECRET
+        },{
+          name: 'USERCONFIG_121_SERVICE_PASSWORD_ADMIN'
+          value: USERCONFIG_121_SERVICE_PASSWORD_ADMIN
+        },{
+          name: 'ENV_ICON'
+          value: ENV_ICON
+        },{
+          name: 'APPLICATION_INSIGHT_ENDPOINT'
+          value: APPLICATION_INSIGHT_ENDPOINT
+        },{
+          name: 'APPLICATION_INSIGHT_IKEY'
+          value: appInsights.properties.InstrumentationKey
+        },{
+          name: 'APPLICATIONINSIGHTS_CONNECTION_STRING'
+          value: 'InstrumentationKey=${appInsights.properties.InstrumentationKey};IngestionEndpoint=https://westeurope-5.in.applicationinsights.azure.com/;LiveEndpoint=https://westeurope.livediagnostics.monitor.azure.com/;ApplicationId=${appInsights.properties.AppId}'
+        },{
+          name: 'ApplicationInsightsAgent_EXTENSION_VERSION'
+          value: ApplicationInsightsAgent_EXTENSION_VERSION
+        },{
+          name: 'ENV_NAME'
+          value: ENV_NAME
+        },{
+          name: 'EXTERNAL_121_SERVICE_URL'
+          value: EXTERNAL_121_SERVICE_URL
```

Comment:
This could maybe also be based on something 'dynamic'? As it is "the thing where the DNS is created for in this script"...